hooks/shapediver/viewer/interaction/useSelectManager.ts

```diff
@@ -0,0 +1,102 @@
+import { ISelectionParameterSettings, MaterialStandardData } from "@shapediver/viewer";
+import { SelectManager, MultiSelectManager, InteractionEngine } from "@shapediver/viewer.features.interaction";
+import { useEffect } from "react";
+import { useInteractionEngine } from "./useInteractionEngine";
+
+// #region Functions (1)
+
+// create an object to store the select managers or the multi select managers for the viewports
+const selectManagers: { [key: string]: {
+	selectManager: SelectManager | MultiSelectManager,
+	selectMultiple: boolean,
+	token: string
+} } = {};
+
+/**
+ * Clean up the select manager for the given viewportId.
+ * We also deselect all selected nodes.
+ *
+ * @param viewportId - The ID of the viewport.
+ * @param interactionEngine - The interaction engine instance.
+ */
+const cleanUpSelectManager = (viewportId: string, interactionEngine?: InteractionEngine) => {
+	if (selectManagers[viewportId]) {
+		if (selectManagers[viewportId].selectManager instanceof SelectManager) {
+			selectManagers[viewportId].selectManager.deselect();
+		} else if (selectManagers[viewportId].selectManager instanceof MultiSelectManager) {
+			selectManagers[viewportId].selectManager.deselectAll();
+		}
+		if(interactionEngine)
+			interactionEngine.removeInteractionManager(selectManagers[viewportId].token);
+		delete selectManagers[viewportId];
+	}
+};
+
+/**
+ * Hook allowing to create a select manager for a viewport.
+ * 
+ * @param viewportId The ID of the viewport.
+ * @param settings The settings for the select manager. If the settings are not provided, the select manager will not be created.
+ */
+export function useSelectManager(viewportId: string, settings?: ISelectionParameterSettings): {
```

Comment:
@MajorMeerkatThe3rd small detail: I suggest to use `Pick< ISelectionParameterSettings, "minimumSelection" | "maximumSelection" | "selectionColor" >` here instead of the complete `ISelectionParameterSettings`